lib/agent.js

```diff
@@ -135,14 +135,11 @@ Agent.prototype.destroy = function () {
   this._instrumentation.stop();
 
   if (this._otelMeterProvider) {
-    // Dev Note: It is unfortunate that `.destroy()` can return while this
-    // shutdown work might still be ongoing. See
-    // https://github.com/elastic/apm-agent-nodejs/issues/3222 to fix that.
-    this._otelMeterProvider
-      .shutdown({ timeoutMillis: 1000 })
-      .catch((reason) => {
-        this.logger.warn('failed to shutdown OTel MeterProvider:', reason);
-      });
+    try {
+      await this._otelMeterProvider.shutdown({ timeoutMillis: 1000 });
```

Comment:
This isn't necessarily the job of *this* change, but here is a concern. This `.shutdown` call, effectively results in await'ing `ElasticApmMetricExporter.flush()` (https://github.com/elastic/apm-agent-nodejs/blob/d86e26a5bb452455550105c226ff1298dabb56db/lib/opentelemetry-metrics/ElasticApmMetricExporter.js#L159-L161), which calls `_agent.flush()`.

At the top of Agent._flush is:

```js
  if (!this._apmClient) {
    // Log an *err* to provide a stack for the user.
    const err = new Error('cannot flush agent before it is started');
    this.logger.warn({ err }, err.message);
    if (cb) {
      process.nextTick(cb);
    }
    return;
  }
```

and earlier in `Agent.destroy` we've done `this._apmClient = null`.   So now I am curious if we are getting this `'cannot flush agent before it is started'` log warning when attempting to `apm.destroy()`. If not, then why not?

So, I think TODOs here are:

1. Add a test for awaiting this `apm.destroy()`.
2. Understand why we are or are not getting that warning.  Note that we will only run this `this._otelMeterProvider.shutdown` if we have instrumented `@opentelemetry/sdk-metrics`, so I'm guessing we just haven't tested this.
3. Likely we want to move the `await this._otelMeterProvider.shutdown(...)` to the *top* of `Agent.prototype.destroy()`.